src/inmanta/module.py

```diff
@@ -107,15 +109,15 @@ def __init__(self, requirement: Requirement) -> None:
     @property
     def project_name(self) -> str:
```

Comment:
I'm inclined to follow `Requirement` here, i.e. add `name` and deprecate `project_name` and `key`. Perhaps make `self._requirement: CanonicalRequirement` (though we probably should not change the constructor argument, we'll have to convert it).
If we don't, we'll need to have another look at the diff here, because chaning `key` to `name` doesn't hold up.

Reply:
I just realized it might hold up because you just don't have the concept of an explicit canonical requirement yet. But this makes it clear to me that there really is a need for it, so we always know what assumptions we can safely make about `r.name`.